lib/api/ldp/ldp-operation.js

```diff
@@ -0,0 +1,143 @@
+'use strict'
+
+const HttpError = require('standard-http-error')
+
+const OPERATIONS = {
+  'head': LdpHeadOperation,
+  'get': LdpGetOperation,
+  'put': LdpPutOperation,
+  // etc
+}
+
+class LdpOperation {
+  /**
+   * @param options {object}
+   *
+   * @param options.target {URL} Parsed full request url
+   *
+   * @param options.headers {object}
+   *
+   * @param [options.authn=null] {object} Authentication object, contains WebID
+   *   string, as well as any bearer credentials/tokens. Needed for
+   *   authenticated fetch (of remote group ACLs, of Copy resources, etc).
+   *   Null if request is not authenticated.
+   *
+   * @param [options.bodyStream] {Stream} Request body stream, for parsing when
+   *   needed
+   */
+  constructor (options) {
+    this.authn = options.authn
+    this.target = options.target
+    this.headers = options.headers
+    this.bodyStream = options.bodyStream
+  }
+
+  static async from ({req}) {
+    const { method, headers } = req
+    const authn = { webId: req.session.webId, credentials: {} } // <- bearer tokens go here
```

Comment:
LDP should definitely not know about this. We should get authentication out of there, and definitely the very specific dependency on a session variable. This kind of state should only be accessed in a single place.

Reply:
Sounds good.